### PR TITLE
Fix unchecked access to Optional

### PR DESCRIPTION
**Pull request by rsynek**

### CODE REVIEW

**Comment by rsynek** [comment]
jenkins retest this

File: optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/persistence/FlightCrewSchedulingXlsxFileIO.java [hunk 1]
```diff
@@ -577,7 +577,9 @@ private void writeEmployeesView() {
                                                 .collect(joining(", ")));
                                 int maxArrivalHour = flightAssignmentList.stream()
                                         .map(a -> a.getFlight().getArrivalUTCTime().getHour())
-                                        .max(Comparator.naturalOrder()).get();
+                                        .max(Comparator.naturalOrder())
+                                        .orElseThrow(() -> new IllegalArgumentException(
+                                                "No flight with maximum arrival hour found."));
```
Comment (ge0ffrey) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/persistence/FlightCrewSchedulingXlsxFileIO.java:582]: Given that there is a `!flightAssignmentList.isEmpty()` check earlier, and that LocalTime.getHour() never returns null (notice that you'd get an NPE if getFlight() or getArrivalUTCTime() returns null), this is an impossible situation (an execution branch that has been theoretically been proven to be impossible). So when an impossible situation happens, it's a bug.
We have the policy to start such exceptions with "Impossible state:".
So I recommend replacing

```
.orElseThrow(() -> new IllegalArgumentException(
                                                "No flight with maximum arrival hour found."));
```

with 

```
.orElseThrow(() -> new IllegalStateException(
                                                "Impossible state: There is not at least one arrival hour in a non-empty flightAssignmentList (" + flightAssignmentList + ")."));
```

Comment (rsynek) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/persistence/FlightCrewSchedulingXlsxFileIO.java:582]: Thanks, I haven't noticed this policy yet.

**Comment by ge0ffrey** [comment]
Thanks @rsynek !

**Pull request merged**